tools/build-release-branch.sh

```diff
@@ -16,117 +17,133 @@
 
 # Exit the build in scary red text if error
 function exit_build {
-    echo -e "${RED}Something went wrong and the build has stopped.  See error above for more details."
-    exit 1
+	echo -e "${RED}Something went wrong and the build has stopped.  See error above for more details."
+	exit 1
 }
 trap 'exit_build' ERR
 
 # Instructions
 function usage {
-    echo "usage: $0 [-n new] [-u update <branchname>]"
-    echo "  -n      Create new release branches"
-    echo "  -u      Update existing release built branch"
-    echo "          Can take an extra param that refers to an existing branch."
-    echo "          Example: $0 -u master-stable"
-    echo "  -h      help"
-    exit 1
+	echo "usage: $0 [-n new] [-u update <branchname>]"
+	echo "  -n      Create new release branches"
+	echo "  -u      Update existing release built branch"
+	echo "          Can take an extra param that refers to an existing branch."
+	echo "          Example: $0 -u master-stable"
+	echo "  -h      help"
+	exit 1
 }
 
 # This creates a new .gitignore file based on master, but removes the items we need for release builds
 function create_release_gitignore {
-    # Copy .gitignore to temp file
-    mv .gitignore .gitignore-tmp
-
-    # Create empty .gitignore
-    touch .gitignore
-
-    # Add things to the new .gitignore file, stopping at the things we want to keep.
-    while IFS='' read -r line || [[ -n "$line" ]]
-    do
-        if [ "$line" == "## Things we will need in release branches" ]
-        then
-            break
-        fi
-        echo "$line" >> .gitignore
-    done < ".gitignore-tmp"
-
-    # Add custom stuff to .gitignore release
-    echo "/_inc/client" >> .gitignore
-
-    # Remove old .gitignore
-    rm .gitignore-tmp
-
-    git commit .gitignore -m "updated .gitignore"
+	# Copy .gitignore to temp file
+	mv .gitignore .gitignore-tmp
+
+	# Create empty .gitignore
+	touch .gitignore
+
+	# Add things to the new .gitignore file, stopping at the things we want to keep.
+	while IFS='' read -r line || [[ -n "$line" ]]; do
+		if [ "$line" == "## Things we will need in release branches" ]; then
+			break
+		fi
+		echo "$line" >> .gitignore
+	done < ".gitignore-tmp"
+
+	# Add custom stuff to .gitignore release
+	echo "/_inc/client" >> .gitignore
+	echo "/docker/" >> .gitignore
+
+	# Needs to stay in sync with .svnignore
+	echo "/extensions/**/*.css" >> .gitignore
+	echo "/extensions/**/*.gif" >> .gitignore
+	echo "/extensions/**/*.jpeg" >> .gitignore
+	echo "/extensions/**/*.jpg" >> .gitignore
+	echo "/extensions/**/*.js" >> .gitignore
+	echo "/extensions/**/*.json" >> .gitignore
+	echo "/extensions/**/*.jsx" >> .gitignore
+	echo "/extensions/**/*.md" >> .gitignore
+	echo "/extensions/**/*.png" >> .gitignore
+	echo "/extensions/**/*.sass" >> .gitignore
+	echo "/extensions/**/*.scss" >> .gitignore
+	echo "/extensions/**/*.svg" >> .gitignore
+	echo "__snapshots__/" >> .gitignore
```

Comment:
Late to the `.gitignore` party, but would a (semi-)whitelist-based approach (i.e. using negation) work? Something like

```
echo "/extensions/" >> .gitignore
echo "!/extensions/**/*.php" >> .gitignore
```

See e.g. https://www.atlassian.com/git/tutorials/saving-changes/gitignore#git-ignore-patterns

Reply:
That would likely work, but would be harder to keep in sync with the `.svnignore` which is really a list of files to pass to `rm -fr`.

Reply:
why we need a `.svnignore` again?

Reply:
It's used in several places around the build/deploy tooling:

https://github.com/Automattic/jetpack/blob/f8078c2cd12ac508334da2fb08e37a92cf283c14/tools/build-jetpack.sh#L70-L75
https://github.com/Automattic/jetpack/blob/58ec3030c9d79e10ace25574cddfaf6d2f50fe6a/tools/build-release-branch.sh#L228-L235
https://github.com/Automattic/jetpack/blob/f8078c2cd12ac508334da2fb08e37a92cf283c14/tools/deploy-to-master-stable-branch.sh#L67-L73
https://github.com/Automattic/jetpack/blob/cadfa61c28b6ed30015313748feb959dfea60235/tools/deploy-to-svn.sh#L72-L74

Reply:
Not for this PR, but I wonder if we could de-dupe sources of truth (i.e. use `.gitifnore` where we're now using `.svnignore`) in the long run... :thinking: